src/main/java/com/nvs/store/service/CartService.java

```diff
@@ -0,0 +1,202 @@
+        return cartRepository
+                .findByUserId(user.getId())
+                .stream()
+                .map(CartUtils.convertToCartDto)
+                .toList();
+    }
+
+    public CartDTO addProductToCart(CartItemRequest cartItemRequest) {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository.findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException(
+                        String.format("User with username %s not found", userEmail))
+                );
+
+        Product product = productRepository
+                .findById(cartItemRequest
+                        .getProductId())
+                .orElseThrow(() ->
+                        new RuntimeException(
+                                String.format("Product with id [%d] does not exist", cartItemRequest.getProductId()))
+                );
+        Cart cart = cartRepository
+                .findByUserIdAndActualTrue(user.getId());
+
+        List<CartItem> cartItems = cartItemRepository.findCartItemByCart(cart);
+        for (CartItem cartItem : cartItems) {
+            if (cartItem.getProduct().getId().equals(product.getId())) {
+                int newQuantity = cartItem.getQuantity() + cartItemRequest.getQuantity();
+                validateAvailableProductQuantities(product, newQuantity);
+                cartItem.setQuantity(newQuantity);
+                cartItemRepository.save(cartItem);
+                return CartUtils.convertToCartDto.apply(cartRepository.findByUserIdAndActualTrue(user.getId()));
+            }
+        }
+
+        validateAvailableProductQuantities(product, cartItemRequest.getQuantity());
+        CartItem cartItem = CartItem
+                .builder()
+                .quantity(cartItemRequest.getQuantity())
+                .product(product)
+                .cart(cart)
+                .build();
+
+        cartItemRepository.save(cartItem);
+
+        return CartUtils.convertToCartDto
+                .apply(cartRepository
+                        .findByUserIdAndActualTrue(user.getId()));
+    }
+
+    public CartDTO orderCart() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository.findByEmail(userEmail).orElseThrow(
+                () -> new RuntimeException(
+                        String.format("User with username %s not found", userEmail))
+        );
+
+        Cart cart = cartRepository
+                .findByUserIdAndActualTrue(user.getId());
+
+        List<CartItem> orderedCartItems = cartItemRepository.findCartItemByCart(cart);
+
+        for (CartItem cartItem : orderedCartItems) {
+            Product product = productRepository.findById(cartItem.getProduct().getId())
+                    .orElseThrow(() -> new RuntimeException(
+                            String.format("Product with id [%d] does not exist", cartItem.getProduct().getId()))
+                    );
+            if (product.getAvailable() < 1) {
```

Comment:
Because of eager fetch loading the product is already loaded together with CartItem. If fetch loading is lazy then you do not have to write sql select to get the product details, you just can call method CartItem#getProduct and hibernate will load the product entity from database without additional select invoked by you.
Please note that a lazy fetch loading or loading product entity by separate select will cause a n+1 select issue.

Reply:
got it 
fixed done ✅